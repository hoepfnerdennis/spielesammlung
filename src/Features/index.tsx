import React, { Fragment } from 'react';
import styles from './styles.module.css';
import { chunkArray } from '../utils';

const Item: React.SFC<{
  icon: string;
  name: string;
  desc: string;
  condition?: boolean;
}> = ({ icon, name, desc, condition = true }) => {
  if (!condition) {
    return null;
  }
  return (
    <p className={styles.players}>
      <span role="img" aria-label={name}>
        {icon}
      </span>
      <b className={styles.highlight}>{desc}</b>
    </p>
  );
};

const List: React.SFC<{
  children: Array<JSX.Element>;
}> = ({ children }) => {
  const filteredChildrenByCondition = children.filter(
    (child) => child.props.condition === undefined || child.props.condition === true
  );
  const chunks = chunkArray<JSX.Element>(filteredChildrenByCondition, 3);
  return (
    <div className={styles.container}>
      {chunks.map((chunk) => (
        <div key={chunk.map((child) => child.props.name).join('_')} className={styles.box}>
          {chunk.map((child) => (
            <Fragment key={child.props.name}>{child}</Fragment>
          ))}
        </div>
      ))}
    </div>
  );
};

export default {
  List,
  Item,
};
