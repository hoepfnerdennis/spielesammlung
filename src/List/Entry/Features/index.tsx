import React from 'react';
import styles from './styles.module.css';

const Features: React.SFC<{
  playersFrom: number;
  playersTo: number;
  age: string;
  duration: string;
  simpleRules: boolean;
}> = ({ playersFrom, playersTo, age, duration, simpleRules }) => {
  return (
    <div className={styles.container}>
      <div className={styles.box}>
        <p className={styles.players}>
          <span role="img" aria-label="Anzahl Spieler">
            👤
          </span>
          <b className={styles.highlight}>
            {playersFrom === playersTo ? playersFrom : `${playersFrom} - ${playersTo}`} Spieler
          </b>
        </p>
        <p className={styles.players}>
          <span role="img" aria-label="Altersempfehlung">
            📅
          </span>
          <b className={styles.highlight}>ab {age} Jahren</b>
        </p>
        <p className={styles.players}>
          <span role="img" aria-label="Spieldauer">
            ⏳
          </span>
          <b className={styles.highlight}>{duration}</b>
        </p>
      </div>
      {simpleRules && (
        <div>
          <p className={styles.players}>
            <span role="img" aria-label="Einfache Regeln">
              🏋️‍♀️
            </span>
            <b className={styles.highlight}>einfach</b>
          </p>
        </div>
      )}
    </div>
  );
};

export default Features;
