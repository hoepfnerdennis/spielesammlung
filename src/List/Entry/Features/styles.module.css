.container {
    display: flex;
}

.box {
    margin-right: 1rem;
}

.players {
    margin: 0 0 .5rem 0;
}

.highlight {
    margin-left: .3rem;
    color: darkcyan;
}
