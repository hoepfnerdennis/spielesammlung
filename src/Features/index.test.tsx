import React from 'react';
import { render } from '@testing-library/react';
import Features from './index';

describe('Features', () => {
  it.each([
    [
      'default',
      {
        playersFrom: 2,
        playersTo: 4,
        age: '4',
        duration: 'duration',
        simpleRules: false,
      },
    ],
    [
      'with simple rules',
      {
        playersFrom: 2,
        playersTo: 2,
        age: '4',
        duration: 'duration',
        simpleRules: true,
      },
    ],
  ])('should render %s', (_, props) => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const { container } = render(<Features {...props} />);
    expect(container).toMatchSnapshot();
  });
});
