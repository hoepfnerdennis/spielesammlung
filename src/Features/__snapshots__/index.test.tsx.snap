// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features should render default 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="box"
    >
      <p
        class="players"
      >
        <span
          aria-label="Anzahl Spieler"
          role="img"
        >
          👤
        </span>
        <b
          class="highlight"
        >
          2 - 4
           Spieler
        </b>
      </p>
      <p
        class="players"
      >
        <span
          aria-label="Altersempfehlung"
          role="img"
        >
          📅
        </span>
        <b
          class="highlight"
        >
          ab 
          4
           Jahren
        </b>
      </p>
      <p
        class="players"
      >
        <span
          aria-label="Spieldauer"
          role="img"
        >
          ⏳
        </span>
        <b
          class="highlight"
        >
          duration
        </b>
      </p>
    </div>
  </div>
</div>
`;

exports[`Features should render with simple rules 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="box"
    >
      <p
        class="players"
      >
        <span
          aria-label="Anzahl Spieler"
          role="img"
        >
          👤
        </span>
        <b
          class="highlight"
        >
          2
           Spieler
        </b>
      </p>
      <p
        class="players"
      >
        <span
          aria-label="Altersempfehlung"
          role="img"
        >
          📅
        </span>
        <b
          class="highlight"
        >
          ab 
          4
           Jahren
        </b>
      </p>
      <p
        class="players"
      >
        <span
          aria-label="Spieldauer"
          role="img"
        >
          ⏳
        </span>
        <b
          class="highlight"
        >
          duration
        </b>
      </p>
    </div>
    <div>
      <p
        class="players"
      >
        <span
          aria-label="Einfache Regeln"
          role="img"
        >
          🏋️‍♀️
        </span>
        <b
          class="highlight"
        >
          einfach
        </b>
      </p>
    </div>
  </div>
</div>
`;
